src/flyte/cli/_common.py

```diff
@@ -316,15 +316,27 @@ def __init__(
     def files(self):
         if self._files is None:
             directory = self._dir or Path(".").absolute()
-            self._files = [os.fspath(p) for p in directory.glob("*.py") if p.name != "__init__.py"]
-            if not self._files:
-                self._files = [os.fspath(".")] + [
-                    os.fspath(p.name) for p in directory.iterdir() if not p.name.startswith(("_", ".")) and p.is_dir()
-                ]
+            # add python files
+            _files = [os.fspath(p) for p in directory.glob("*.py") if p.name != "__init__.py"]
+
+            # add directories
+            _files.extend([os.fspath(".")] + [
+                os.fspath(directory / p.name) for p in directory.iterdir() if not p.name.startswith(("_", ".")) and p.is_dir()
+            ])
+            self._files = _files
         return self._files
 
     def list_commands(self, ctx):
-        return self.files
+        file_commands = []
+        for f in self.files:
+            if Path(f).is_relative_to(Path.cwd()):
```

Comment:
thank you. but we could do this in the files itself? that will be faster, as the self.files is cached